gcloud-java-compute/src/main/java/com/google/cloud/compute/Operation.java

```diff
@@ -0,0 +1,846 @@
+     */
+    public String code() {
+      return code;
+    }
+
+    /**
+     * Returns a human-readable error message.
+     */
+    public String message() {
+      return message;
+    }
+
+    /**
+     * Returns metadata about this warning. Each key provides more detail on the warning being
+     * returned. For example, for warnings where there are no results in a list request for a
+     * particular zone, this key might be {@code scope} and the key's value might be the zone name.
+     * Other examples might be a key indicating a deprecated resource, and a suggested replacement,
+     * or a warning about invalid network settings (for example, if an instance attempts to perform
+     * IP forwarding but is not enabled for IP forwarding).
+     */
+    public Map<String, String> metadata() {
+      return metadata;
+    }
+
+    com.google.api.services.compute.model.Operation.Warnings toPb() {
+      com.google.api.services.compute.model.Operation.Warnings warningPb =
+          new com.google.api.services.compute.model.Operation.Warnings()
+              .setCode(code)
+              .setMessage(message);
+      if (this.metadata != null) {
+        List<com.google.api.services.compute.model.Operation.Warnings.Data> metadataPb =
+            Lists.newArrayListWithCapacity(metadata.size());
+        for (Map.Entry<String, String> entry : metadata.entrySet()) {
+          metadataPb.add(new com.google.api.services.compute.model.Operation.Warnings.Data()
+              .setKey(entry.getKey()).setValue(entry.getValue()));
+        }
+        warningPb.setData(metadataPb);
+      }
+      return warningPb;
+    }
+
+    static OperationWarning fromPb(
+        com.google.api.services.compute.model.Operation.Warnings warningPb) {
+      Map<String, String> metadata = null;
+      if (warningPb.getData() != null) {
+        metadata = Maps.newHashMapWithExpectedSize(warningPb.getData().size());
+        for (com.google.api.services.compute.model.Operation.Warnings.Data data
+            : warningPb.getData()) {
+          metadata.put(data.getKey(), data.getValue());
+        }
+      }
+      return new OperationWarning(warningPb.getCode(), warningPb.getMessage(), metadata);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      return obj instanceof OperationWarning
+          && Objects.equals(code, ((OperationWarning) obj).code)
+          && Objects.equals(message, ((OperationWarning) obj).message)
+          && Objects.equals(metadata, ((OperationWarning) obj).metadata);
+    }
+
+    @Override
+    public String toString() {
+      return MoreObjects.toStringHelper(this)
+          .add("code", code)
+          .add("message", message)
+          .add("metadata", metadata)
+          .toString();
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(code, message, metadata);
+    }
+  }
+
+  static final class Builder {
+
+    private Compute compute;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Private field 'compute' could be made final; it is only initialized in the declaration or constructor.](https://www.codacy.com/app/mziccard/gcloud-java/file/2746909904/issues/source?bid=3123107&fileBranchId=3356751#l302)